scorecard/scoring.py

```diff
@@ -1,39 +1,122 @@
 from django.db.models import Q
-from .models import Product, ProductControl
+from .models import BusinessUnit, Product, ProductControl, ProductSecurityCapability, ProductSecurityRole, \
+    SecurityCapability, SecurityRole, Status
 
 
-# potential statuses:
-#       Implemented, complete   (green)     3
-#       partial                 (yellow)    2
-#       planned                 (orange)    1
-#       none, unknown           (red)       0
-#       not applicable          (grey)      -
+'''
+    Executes all scoring related functionality.
+
+    potential statuses:
+          Implemented, complete   (green)     3
+          partial                 (yellow)    2
+          planned                 (orange)    1
+          none, unknown           (red)       0
+          not applicable          (grey)      -    
+'''
 
 
 def get_control_status_values():
     return {'Implemented': 3, 'complete': 3, 'partial': 2, 'planned': 1, 'none': 0, 'unknown': 0}
 
 
-def get_product_score(product):
+def get_product_scores(product):
+
+    # Calculate People score.
+    people_score = 0
+    security_roles = SecurityRole.objects.all()
+    max_people_score = len(security_roles) # 1 point for each security role.
+    prod_sec_roles = ProductSecurityRole.objects.filter(product=product)
+    for security_role in security_roles:
+        for prod_sec_role in prod_sec_roles:
+            if prod_sec_role.role == security_role and prod_sec_role.person is not None:
+                people_score += 1
+                break
+
+    # Calculate Process score.
+    process_score = 0
+    max_process_score = 1
+
+    # Calculate Technology score.
+    technology_score = 0
+    security_capabilities = SecurityCapability.objects.all()
+    product_security_capabilities = ProductSecurityCapability.objects.filter(Q(product=product) & ~Q(status__value=-1))
+    max_technology_score = len(security_capabilities) * get_max_status_value()
+    for product_security_capability in product_security_capabilities:
+        technology_score += product_security_capability.status.value
+
+    # Calculate Compliance score.
+    compliance_score = 0
     status_values = get_control_status_values()
     # Get the product controls, but filter out all those with a status of 'not applicable'.
     product_controls = ProductControl.objects.filter(Q(product=product) & ~Q(status='not applicable')).\
         order_by('control__name')
-    max_score = len(product_controls) * 3  # A maximum possible score is 3 x the number of controls.
-    score = 0
+    # max_compliance_score = len(product_controls) * 3  # A maximum possible score is 3 x the number of controls.
+    max_compliance_score = 1
     for product_control in product_controls:
         try:
-            score += status_values[product_control.status]
+            compliance_score += status_values[product_control.status]
         except KeyError:
             print(f">>>> ERROR: Unknown product control status: '{product_control.status}'")
```

Comment:
IIRC. This was just introduced in python3.6. Please make this more friendly by doing the old style.